### PR TITLE
install path fixes

### PR DESCRIPTION
**Pull request by c0d3z3r0**
### Motivation and Context

This PR fixes some install paths.

### Description

rpm: correct pkgconfig path
pkconfig files get installed to $datarootdir/pkgconfig but rpm expects
them to be at $datadir. This works when $datarootdir==$datadir which is
the case most of the time but will fail when they differ.

install: make initramfs-tools path static
Since initramfs-tools' path is nothing we can control as it is an
external package it does not make any sense to install zfs additions
anywhere else. Installing it elsewhere wouldn't even work since
mkinitramfs does only honor /etc/initramfs-tools and /usr/share/initramfs-tools.
Simply use /usr/share/initramfs-tools as path.

### How Has This Been Tested?

Tested locally with different configurations.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by c0d3z3r0** [comment]
That one failing test does not seem to be related to my changes:
~~~
ln: failed to create symbolic link '/var/lib/buildbot/slaves/zfs/Fedora_30_x86_64__TEST_/build/tests/full.log': File exists
+ zfs-tests.sh -vx -d /mnt/ -r linux -I 1 -T functional
+ RESULT=1
+ set +x
tar: zts-linux-20190729T071953/core: Cannot open: Permission denied
tar: Exiting with failure status due to previous errors
~~~

File: contrib/initramfs/Makefile.am [hunk 1]
```diff
@@ -1,4 +1,4 @@
-initrddir = $(datarootdir)/initramfs-tools
+initrddir = /usr/share/initramfs-tools
```
Comment (behlendorf) [inline contrib/initramfs/Makefile.am]: Just one question, do we know of any distributions when this external package installs itself somewhere else?  Would it be worthwhile to provide a mechanism to set this path via configure?

Comment (c0d3z3r0) [reply contrib/initramfs/Makefile.am]: I am not aware of any distribution having anything else than `/usr/share`.

Debian does hard code it (see https://salsa.debian.org/kernel-team/initramfs-tools/blob/master/mkinitramfs), ubuntu does not change this. I could not find any debian based distribution with any other path

**Comment by rlaager** [comment]
LGTM

**Comment by grahamc** [comment]
It seems this has broken the NixOS build (https://github.com/zfsonlinux/zfs/pull/9161#issuecomment-529196847) I wonder why not go for `$datadir`?

**Comment by c0d3z3r0** [comment]
@grahamc what exactly do you mean by „break“? See description for why I changed this

**Comment by Mic92** [comment]
@c0d3z3r0 we don't install software to `/usr` in NixOS instead we install all software into `/nix/store/<sha256sum>-zfs-<version>` so we can achieve our design goals: https://nixos.org/nix/

**Comment by c0d3z3r0** [comment]
@Mic92 ah, thank you! Ok, then it‘s a packaging problem on Nix side pretty sure

**Comment by rlaager** [comment]
Nix is a package manager so it is not using the RPM spec file, correct? So the issue is the other part, the change from $datarootdir to /usr/share for initramfs-tools? If so, what path is NixOS using for initramfs-tools today?

**Comment by c0d3z3r0** [comment]
Oh wait, I was a bit to fast... yes, this is a ZFS issue as the path is hard coded now... oops...

**Comment by c0d3z3r0** [comment]
@Mic92 @rlaager @grahamc https://github.com/zfsonlinux/zfs/pull/9344

**Comment by Mic92** [comment]
@rlaager we do not have this package. However if we would want to `initramfs-tools` to find this directory, we would either zfs add to our system packages, which would add symlinks for this directory to `/run/current-system/sw/share` or we would introduce some sort of shell wrapper. RPM spec file is not used in NixOS.

**Pull request merged**